File: 070ClimbingStairs.cpp
class Solution {
public:
    // 用于记录已经算过的数值，避免重复计算
    int a[1000];

    int climbStairs(int n) {
        if(n <= 2)
            return n;
        
        int nmin1 = 0;
        int nmin2 = 0;
        if(a[n-1] != 0){
            nmin1 = a[n-1];
        }else{
            nmin1 = climbStairs(n-1);
            a[n-1] = nmin1;
        }
        if(a[n-2] != 0){
            nmin2 = a[n-2];
        }else{
            nmin2 = climbStairs(n-2);
            a[n-2] = nmin2;
        }
        
        return nmin1 + nmin2;
    }
};

File: 070ClimbingStairs.py
class Solution:
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        results = [i for i in range(n+1)]
        if n == 1 or n == 2:
            return results[n]
        else:
            for i in range(len(results)):
                if i >= 3:
                    results[i] = results[i-1] + results[i-2]
            return results[n]
